self-hosting/GUIDE.md

```diff
@@ -0,0 +1,375 @@
+
+When the service is ready you can access the web-app by using localhost:3000
+
+With the default .env file you will be able to use Omnivore, add RSS Feeds, add stories etc. 
+
+
+### Additional Services used: 
+
+#### PGVector
+A PGVector image is used to provide Postgres functionality. If you have another postgres service running it is possible to remove 
+this from the docker-compose and provide the host, username and password of the Postgres instance. 
+
+#### Redis 
+Redis is used as a queueing system, and for caching. If you have a Redis Instance already it is possible to remove this from the docker-compose 
+and rely on the hosted Redis. You must replace the redis url for this. 
+
+#### Minio (Self-Host)
+Minio is an AWS S3 compatible Object storage service that you can self-host. It is included in the docker-compose file. 
+It allows you to use the S3 Storage API. 
+
+We also have a small client that creates the necessary buckets (createbuckets). See below: 
+```bash
+until (/usr/bin/mc config host add myminio http://minio:9000 minio miniominio) do echo '...waiting...' && sleep 1; done;
+/usr/bin/mc mb myminio/omnivore;
+/usr/bin/mc policy set public myminio/omnivore;
+```
+
+If you use GCS, or S3 buckets you can do the following:
+
+##### S3 (Optional): 
+S3 is an AWS Block Storage Service. You can also use S3 as your storage service, rather than the included MinIO self-host. In order to use S3, you must do the following.
+
+Replace the following with the correct parameters. 
+```env
+AWS_ACCESS_KEY_ID=minio   # Used for Minio S3 Client
+AWS_SECRET_ACCESS_KEY=miniominio
+AWS_REGION=us-east-1
+```
+
+Replace the following with an endpoint URL from [here](https://docs.aws.amazon.com/general/latest/gr/s3.html)
+```env
+LOCAL_MINIO_URL=http://localhost:1010
+
+```
+
+##### GCS (Optional): 
+Remove the following Environment Variable: 
+```env
+GCS_USE_LOCAL_HOST=true
+```
+and populate 
+```
+GCS_UPLOAD_SA_KEY_FILE_PATH
+```
+with the path of the JSON key file for the service account.
+
+## Nginx Reverse Proxy
+
+Nginx is a reverse proxy that receives requests, and directs them to the correct service internally. Omnivore runs 4 services we want to redirect to.
+
+* Omnivore Web
+* Omnivore API
+* Omnivore Bucket [Optional]
+* Omnivore Image Proxy [Optional]
+
+We have included an example Nginx Configuration that redirects traffic from http (80) to https (443), and then directs traffic to the correct service based on the request path. 
+
+```nginx
+events {}
+
+http {
+    sendfile    on;
+    keepalive_timeout 60;
+
+     upstream omnivore_web {
+    	ip_hash;
+    	server 127.0.0.1:3000;
+     }
+
+     upstream omnivore_backend {
+    	ip_hash;
+    	server 127.0.0.1:4000;
+     }
+
+     upstream omnivore_imageproxy {
+    	ip_hash;
+    	server 127.0.0.1:1010;
+     }
+
+    upstream omnivore_bucket {
+    	ip_hash;
+    	server 127.0.0.1:7070;
+    }
+
+    server {
+        listen 80;
+        return 301 https://$host$request_uri
```

Comment:
Could you sync this example .conf with your updated version in nginx/nginx.conf? For example, you added a semicolon on this line to fix a bug but the readme still has the bug present.

Reply:
There are quite a few changes. It's probably easier to just link to the file itself vs trying to keep this up to date.